app/components/ui/table.tsx

```diff
@@ -0,0 +1,63 @@
+import { useTable, Column } from "react-table";
+
+interface TableProps<T extends object> {
+	columns: Column<T>[];
+	data: T[];
+}
+
+export const Table = <T extends object>({ columns, data }: TableProps<T>) => {
+	const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable<T>({
+		columns,
+		data,
+	});
+	const isTableEmpty = data.length === 0;
+	const renderNoDataMessage = (
+		<tr>
+			<td colSpan={columns.length} className="p-3 text-center text-[13px] text-[#FFFFFFCC]">
+				No data available
+			</td>
+		</tr>
+	);
+
+	return (
+		<div className="overflow-x-auto">
+			<table {...getTableProps()} className="w-full text-left bg-azure-10 rounded-3xl">
+				<thead>
+					{headerGroups.map((headerGroup) => (
+						<tr
+							{...headerGroup.getHeaderGroupProps()}
+							key={headerGroup.id}
+							className="text-[#FFFFFFCC] font-medium text-[13px]"
+						>
+							{headerGroup.headers.map((column) => (
+								<th {...column.getHeaderProps()} key={column.id} className="p-3">
+									{column.render("Header")}
+								</th>
+							))}
+						</tr>
+					))}
+				</thead>
```

Comment:
let's move this to a separate function, to make this function shorter.

Reply:
Done.